cli/azd/cmd/provision.go

```diff
@@ -159,6 +163,32 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 	}
 	defer func() { _ = infra.Cleanup() }()
 
+	wroteNewInput := false
+
+	for inputName, inputInfo := range infra.Inputs {
+		inputConfigKey := fmt.Sprintf("inputs.%s", inputName)
+
+		if _, has := p.env.Config.GetString(inputConfigKey); !has {
+			// No value found, so we need to generate one, and store it in the config bag.
+			val, err := password.FromAlphabet(password.LettersAndDigits, *inputInfo.Default.Generate.MinLength)
```

Comment:
I think we only assume `inputs` have these sections attached for now, and simply panic if unsupported?

Reply:
Yeah - corners cut here due to the enforcement here:

https://github.com/Azure/azure-dev/pull/3131/files#diff-b711e594f7ac2f56ef4386a9fad07d756710f296fc0a2496c3148f6ec0c1b821R283-R293

Which validates that ever input can be generated today and has a min length.  There's a TODO around that code, but I'll add some notes here as well as to why this dereference is safe today and how it will need to be improved when we relax this requirement.

Reply:
Added some load bearing TODO's...